crates/router/src/core/webhooks/outgoing_v2.rs

```diff
@@ -0,0 +1,817 @@
+                    )
+                    .await?;
+                }
+            }
+        },
+        // TODO: Add support for automatic retries
+        enums::WebhookDeliveryAttempt::AutomaticRetry => todo!(),
+        enums::WebhookDeliveryAttempt::ManualRetry => match response {
+            Err(client_error) => {
+                api_client_error_handler(
+                    state.clone(),
+                    merchant_key_store.clone(),
+                    &business_profile.merchant_id,
+                    &event_id,
+                    client_error,
+                    delivery_attempt,
+                    ScheduleWebhookRetry::NoSchedule,
+                )
+                .await?
+            }
+            Ok(response) => {
+                let status_code = response.status();
+                let _updated_event = update_event_in_storage(
+                    state.clone(),
+                    merchant_key_store.clone(),
+                    &business_profile.merchant_id,
+                    &event_id,
+                    response,
+                )
+                .await?;
+
+                if status_code.is_success() {
+                    increment_webhook_outgoing_received_count(&business_profile.merchant_id);
+                } else {
+                    error_response_handler(
+                        state,
+                        &business_profile.merchant_id,
+                        delivery_attempt,
+                        status_code.as_u16(),
+                        "Ignoring error when sending webhook to merchant",
+                        ScheduleWebhookRetry::NoSchedule,
+                    )
+                    .await?;
+                }
+            }
+        },
+    }
+
+    Ok(())
+}
+
+async fn raise_webhooks_analytics_event(
+    state: SessionState,
+    trigger_webhook_result: CustomResult<(), errors::WebhooksFlowError>,
+    content: Option<api::OutgoingWebhookContent>,
+    merchant_id: common_utils::id_type::MerchantId,
+    event: domain::Event,
+    merchant_key_store: &domain::MerchantKeyStore,
+) {
+    let key_manager_state: &KeyManagerState = &(&state).into();
+    let event_id = event.event_id;
+
+    let error = if let Err(error) = trigger_webhook_result {
+        logger::error!(?error, "Failed to send webhook to merchant");
+
+        serde_json::to_value(error.current_context())
+            .change_context(errors::ApiErrorResponse::WebhookProcessingFailure)
+            .inspect_err(|error| {
+                logger::error!(?error, "Failed to serialize outgoing webhook error as JSON");
+            })
+            .ok()
+    } else {
+        None
+    };
+
+    let outgoing_webhook_event_content = content
+        .as_ref()
+        .and_then(api::OutgoingWebhookContent::get_outgoing_webhook_event_content)
+        .or_else(|| get_outgoing_webhook_event_content_from_event_metadata(event.metadata));
+
+    // Fetch updated_event from db
+    let updated_event = state
```

Comment:
Why we are find the event again since the event is already in scope in the flow?